pkg/k8s/watchers/cilium_endpoint_slice.go

```diff
@@ -22,6 +25,34 @@ var (
 	cepMap = newCEPToCESMap()
 )
 
+// CreateLocalNodeIndexFunc returns an IndexFunc that can index CiliumEndpointSlice objects
+// by the specified nodeIP (i.e. you can use this maintain an index of CES with endpoints
+// referencing the local node).
+// If nodeIP is empty, this will simply maintain an index of all referenced nodes.
+func CreateLocalNodeIndexFunc(nodeIP string) cache.IndexFunc {
+	return func(obj interface{}) ([]string, error) {
+		ces, ok := obj.(*v2alpha1.CiliumEndpointSlice)
+		if !ok {
+			return nil, fmt.Errorf("unexpected object type: %T", obj)
+		}
+		indices := []string{}
+		for _, ep := range ces.Endpoints {
+			if nodeIP == "" {
+				indices = append(indices, ep.Networking.NodeIP)
+			} else {
+				if ep.Networking.NodeIP == nodeIP {
+					// If we're only indexing a particular nodeIP, then as soon as we
```

Comment:
Nit: 

```
else if ... {
```

rather than another indented block.

Reply:
@tommyp1ckles doesn't seem resolved. This refers to the second commit.

Reply:
@aanm right thanks, I need to squash down some changes I think, fixing this just gets clobbered by a conflict down the line.